src/entitysdk/exception.py

```diff
@@ -19,3 +22,55 @@ class DependencyError(EntitySDKError):
 
 class StagingError(EntitySDKError):
     """Raised when a staging operation has failed."""
+
+
+class ServerError(EntitySDKError):
+    """Raises when a server request error is encountered."""
+
+    def __init__(self, *, response, message):
+        """Store response for server error."""
+        request = response.request
+        try:
+            json_response_data = response.json()
+        except (JSONDecodeError, UnicodeDecodeError):
+            json_response_data = None
+
+        text_response_data = response.text
+
+        try:
+            json_request_data = loads(request.content.decode())
+        except JSONDecodeError:
+            json_request_data = request.content.decode()
+
+        text_request_data = request.content.decode("utf-8")
+
+        self.details = {
+            "request": {
+                "method": request.method,
+                "url": str(request.url),
+                "text": text_request_data,
+                "json": json_request_data,
+            },
+            "response": {
+                "status_code": response.status_code,
+                "text": text_response_data,
+                "json": json_response_data,
+            },
+        }
+
+        super().__init__(message)
+
+    def __str__(self):
+        """Return string representation of the error details."""
+        # for printing do not include empty entries
+        message_summary = {"message": self.args[0]} | {
+            transaction: {k: v for k, v in info.items() if v}
+            for transaction, info in self.details.items()
+        }
+        # and if there is a json, do not show text for less noise
+        for transaction in ("request", "response"):
+            if message_summary[transaction].get("text") and message_summary[transaction].get(
+                "json"
+            ):
+                del message_summary[transaction]["text"]
+        return dumps(message_summary, indent=2)
```

Comment:
Since the str representation doesn't need to be a valid json dump, what about using the message as a presentation of the error instead of adding it to the payload? 
Something like:
```
Server request failed:
{
  "request": { ...},
  "response": {...}
}
```

Can we also add a test to check the str representation of the exception?